### PR TITLE
coot/threadDelay fix

### PR DESCRIPTION
**Pull request by coot**
- Fixed a bug in threadDelay IO instance
- Added a TODO note

### CODE REVIEW

**Comment by coot** [comment]
bors merge

File: io-sim-classes/src/Control/Monad/Class/MonadTimer.hs [hunk 1]
```diff
@@ -201,6 +201,10 @@ instance MonadTimer IO where
   -- | For delays less (or equal) than @maxBound :: Int@ this is exactly the same as
   -- 'STM.registerDaley'; for larger delays it will start a monitoring thread
   -- whcih will update the 'TVar'.
+  --
+  -- TODO: issue #2184 'registerDelay' relayes on `newTimeout`, through
```
Comment (mrBliss) [inline io-sim-classes/src/Control/Monad/Class/MonadTimer.hs:205]:
```suggestion
  -- TODO: issue #2184 'registerDelay' relies on `newTimeout`, through
```

**Comment by coot** [comment]
bors merge

**Pull request merged**